src/Compilers/CSharp/Portable/Symbols/TypeUnification.cs

```diff
@@ -19,28 +17,28 @@ internal static class TypeUnification
         /// </summary>
         public static bool CanUnify(TypeSymbol t1, TypeSymbol t2)
         {
-            if (TypeSymbol.Equals(t1, t2, TypeCompareKind.ConsiderEverything2))
+            if (TypeSymbol.Equals(t1, t2, TypeCompareKind.IgnoreNativeIntegers))
```

Comment:
>IgnoreNativeIntegers [](start = 58, length = 20)

Should we also ignore nullability in this component? Would it make sense to use CLRSignatureCompareOptions instead? Could be more robust long term.

Reply:
Makes sense, thanks.

---
In reply to: [500383811](https://github.com/dotnet/roslyn/pull/48349#discussion_r500383811) [](ancestors = 500383811)